src/SonarScanner.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -39,26 +39,11 @@ namespace SonarScanner.MSBuild.Tasks;
 /// <remarks>The task does not make any assumptions about the type of project from which it is
 /// being called so it should work for projects of any type - C#, VB, UML, C++, and any new project types
 /// that are created.</remarks>
-public class WriteProjectInfoFile : Task
+public class WriteProjectInfoFile(IEncodingProvider encodingProvider) : Task
```

Comment:
All the changes in these files are refactorings and should not change the behavior:
- used primary constructor
- removed namespaces
- reordered methods to follow our conventions
- removed `GlobalAnalysisSettings` property that was not used
- simplified `TryCreateAnalysisResults` and `TryCreateAnalysisSettings` methods by using Linq
- reduced nesting in `TryCreateResultFromItem`
- used `is null` and `is not null` where applicable